frontend/device/generic/device.lua

```diff
@@ -196,4 +196,27 @@ function Device:exit()
     self.screen:close()
 end
 
+function Device:retrieveNetworkInfo()
+    local std_out = io.popen("ifconfig | " ..
+                             "grep -P 'Link|addr' | " ..
```

Comment:
Have you tested it? On my system your command fails, because grep has no '-P' option, look:

```
 ifconfig | grep -P 'Link|addr'
```

This is even without the rest of your long command (which I think can be shortened to fewer processes).